src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Unix.cs

```diff
@@ -284,12 +290,19 @@ private IAsyncResult BeginSendFileInternal(string? fileName, byte[]? preBuffer,
             // Open it before we send the preBuffer so that any exception happens first
             FileStream? fileStream = OpenFile(fileName);
 
-            return TaskToApm.Begin(SendFileInternalAsync(fileStream, preBuffer, postBuffer), callback, state);
+            return TaskToApm.Begin(SendFileInternalAsync(fileStream, preBuffer, postBuffer).AsTask(), callback, state);
         }
 
         private void EndSendFileInternal(IAsyncResult asyncResult)
         {
             TaskToApm.End(asyncResult);
         }
+
+        internal sealed class FileSendSocketAsyncEventargs : SocketAsyncEventArgs, IValueTaskSource
```

Comment:
Unfortunately this one is needed here too, to allow https://github.com/dotnet/runtime/blob/cfebea2949f44af5ae240bc67282d3e35fbfe058/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs#L594